test/lib/redis-store-spec.js

```diff
@@ -463,3 +466,39 @@ describe('defaults', function () {
   });
 });
 
+describe('wrap function', function () {
+
+  // Simulate retrieving a user from a database
+  function getUser(id, cb) {
+    setTimeout(function () {
+      cb(null, {id: id, name: 'Bob'});
+    }, 100);
+  }
+
+  var userId = 123;
+  var key = 'user_' + userId;
+  var fromCache = true;
+
+  it('should work with wrap function', function (done) {
+
+    // First call to wrap should run the code
+    redisCache.wrap(key, function (cb) {
+      fromCache = false;
+      getUser(userId, cb);
+    }, function (err, user) {
+      assert.equal(fromCache, false);
+      assert.equal(user.name, 'Bob');
+
+      // Second call to wrap should retrieve from cache
+      fromCache = true;
+      redisCache.wrap(key, function (cb) {
+        fromCache = false;
+        getUser(userId, cb);
+      }, function (err, user) {
+        assert.equal(user.name, 'Bob');
+        assert.equal(fromCache, true);
```

Comment:
it is true because you set it to true on line 493 (or 480) and not because it is cached.